### PR TITLE
planner: update cost model2

### PR DESCRIPTION
**Pull request by qw4990**
### What problem does this PR solve?

Issue Number: ref #35240

Problem Summary: planner: update cost model2

### What is changed and how it works?

planner: update cost model2
1. add some start-costs to HashJoin/IndexJoin to let the optimizer prefers to use MergeJoin if data size is small;
2. add some start-costs to HashAgg to let the optimizer prefers to use StreamAgg and not push HashAgg down if the data-size is small;

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: executor/explain_test.go [hunk 1]
```diff
@@ -364,7 +364,7 @@ func TestCheckActRowsWithUnistore(t *testing.T) {
 		},
 		{
 			sql:      "select count(*) from t_unistore_act_rows group by b",
-			expected: []string{"2", "2", "2", "4"},
+			expected: []string{"2", "4", "4"},
```
Comment (qw4990) [inline executor/explain_test.go:367]: Expected, not push Agg down.

File: executor/explainfor_test.go [hunk 2]
```diff
@@ -462,12 +462,12 @@ func TestPointGetUserVarPlanCache(t *testing.T) {
 	tk.Session().SetSessionManager(&testkit.MockSessionManager{PS: ps})
 	tk.MustQuery(fmt.Sprintf("explain for connection %d", tkProcess.ID)).Check(testkit.Rows( // can use idx_a
 		`Projection_9 1.00 root  test.t1.a, test.t1.b, test.t2.a, test.t2.b`,
-		`└─IndexJoin_17 1.00 root  inner join, inner:TableReader_13, outer key:test.t2.a, inner key:test.t1.a, equal cond:eq(test.t2.a, test.t1.a)`,
-		`  ├─Selection_44(Build) 0.80 root  not(isnull(test.t2.a))`,
-		`  │ └─Point_Get_43 1.00 root table:t2, index:idx_a(a) `,
-		`  └─TableReader_13(Probe) 0.00 root  data:Selection_12`,
-		`    └─Selection_12 0.00 cop[tikv]  eq(test.t1.a, 1)`,
-		`      └─TableRangeScan_11 0.80 cop[tikv] table:t1 range: decided by [eq(test.t1.a, test.t2.a)], keep order:false, stats:pseudo`))
+		`└─MergeJoin_10 1.00 root  inner join, left key:test.t2.a, right key:test.t1.a`,
```
Comment (qw4990) [inline executor/explainfor_test.go:465]: Expected, MergeJoin is more cpu-effective if data-size is small.

File: executor/join_test.go [hunk 3]
```diff
@@ -1479,7 +1479,7 @@ func TestIndexNestedLoopHashJoin(t *testing.T) {
 		"      └─TableRowIDScan 27.00 cop[tikv] table:l2 keep order:false"))
 	tk.MustQuery("select * from t l1 where exists ( select * from t l2 where l2.l_orderkey = l1.l_orderkey and l2.l_suppkey <> l1.l_suppkey )order by `l_orderkey`,`l_linenumber`;").Check(testkit.Rows("0 0 0 0", "0 1 0 1", "0 2 0 0", "1 0 1 0", "1 1 1 1", "1 2 1 0", "2 0 0 0", "2 1 0 1", "2 2 0 0"))
 	tk.MustQuery("desc format = 'brief' select count(*) from t l1 where exists ( select * from t l2 where l2.l_orderkey = l1.l_orderkey and l2.l_suppkey <> l1.l_suppkey );").Check(testkit.Rows(
-		"HashAgg 1.00 root  funcs:count(1)->Column#11",
+		"StreamAgg 1.00 root  funcs:count(1)->Column#11",
```
Comment (qw4990) [inline executor/join_test.go:1482]: Expected, StreamAgg is more cpu-effective if data-size is small.

File: planner/core/testdata/binary_plan_suite_out.json [hunk 4]
```diff
@@ -123,23 +123,23 @@
                         "operator_info": "data:TableFullScan_16"
                       }
                     ],
-                    "cost": 975351.9825195674,
```
Comment (qw4990) [inline planner/core/testdata/binary_plan_suite_out.json:126]: Expected, just update cost values.

File: planner/core/testdata/expression_rewriter_suite_out.json [hunk 5]
```diff
@@ -7,10 +7,9 @@
         "Plan": [
           "HashJoin 2.25 root  inner join, equal:[eq(test.t1.a, test.t2.a) eq(test.t1.b, test.t2.b)]",
           "├─HashAgg(Build) 1.69 root  group by:test.t2.a, test.t2.b, funcs:firstrow(test.t2.a)->test.t2.a, funcs:firstrow(test.t2.b)->test.t2.b",
-          "│ └─TableReader 1.69 root  data:HashAgg",
-          "│   └─HashAgg 1.69 cop[tikv]  group by:test.t2.a, test.t2.b, ",
```
Comment (qw4990) [inline planner/core/testdata/expression_rewriter_suite_out.json:11]: Expected, not push Agg down if data-size is small.

**Review by chrysan — APPROVED** [review]

**Review by time-and-fate — APPROVED** [review]

File: planner/core/plan_cost_ver2.go [hunk 6]
```diff
@@ -448,13 +451,16 @@ func (p *PhysicalHashAgg) getPlanCostVer2(taskType property.TaskType, option *Pl
 	groupCost := groupCostVer2(option, inputRows, p.GroupByItems, cpuFactor)
 	hashBuildCost := hashBuildCostVer2(option, outputRows, outputRowSize, float64(len(p.GroupByItems)), cpuFactor, memFactor)
 	hashProbeCost := hashProbeCostVer2(option, inputRows, float64(len(p.GroupByItems)), cpuFactor)
+	startCost := newCostVer2(option, cpuFactor,
+		10*3*cpuFactor.Value,
```
Comment (time-and-fate) [inline planner/core/plan_cost_ver2.go:455]: Please add some comments to explain the `10*3`.

Comment (qw4990) [reply planner/core/plan_cost_ver2.go]: updated

**Pull request merged**